src/main/java/com/google/sps/data/Election.java

```diff
@@ -14,81 +14,166 @@
 @AutoValue
 public abstract class Election {
-  public abstract long getId();
+  public abstract String getId();
 
   public abstract String getName();
 
   public abstract String getDate();
 
   public abstract String getScope();
 
-  public abstract HashSet<Long> getContests();
+  public abstract HashSet<String> getContests();
 
-  public abstract HashSet<Long> getPropositions();
+  public abstract HashSet<String> getPropositions();
 
   public static Builder builder() {
     return new AutoValue_Election.Builder();
   }
 
-  // creates a new Election object by extracting the properties from "obj"
-  // TODO(caseyprice): Update setContests and setPropositions parameters
-  public static Election fromJSONObject(JSONObject obj) throws JSONException {
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder setId(String id);
+
+    public abstract Builder setName(String name);
+
+    public abstract Builder setScope(String scope);
+
+    public abstract Builder setDate(String date);
+
+    public abstract Builder setContests(HashSet<String> contests);
+
+    public abstract Builder setPropositions(HashSet<String> propositions);
+
+    public abstract Election build();
+  }
+
+  /**
+   * Creates a new Election object by extracting the properties from "electionQueryData".
+   *
+   * @param electionQueryData the JSON output of an electionQuery to the Google Civic Information
+   *     API
+   * @return the new Election object
+   */
+  public static Election fromElectionQuery(JSONObject electionQueryData) throws JSONException {
     return Election.builder()
-        .setId(obj.getLong("id"))
-        .setName(obj.getString("name"))
-        .setDate(obj.getString("electionDay"))
-        .setScope(obj.getString("ocdDivisionId"))
-        // TODO(caseyprice): get values for Contests and Propositions
-        .setContests(new HashSet<Long>())
-        .setPropositions(new HashSet<Long>())
+        .setId(electionQueryData.getString("id"))
+        .setName(electionQueryData.getString("name"))
+        .setDate(electionQueryData.getString("electionDay"))
+        .setScope(electionQueryData.getString("ocdDivisionId"))
+        .setContests(new HashSet<String>())
+        .setPropositions(new HashSet<String>())
         .build();
   }
 
-  // creates a new Entity and sets the proper properties.
-  public Entity toEntity() {
+  /**
+   * Creates a new Election object by extracting the properties from an Election object and from the
+   * properties of "voterInfoQueryData". Delegates creating Contest Entities from this election's
+   * list of contests.
+   *
+   * @param election the Election object already populated by fromElectionQuery
+   * @param datastore the Datastore to store this election's list of contests
+   * @param voterInfoQueryData the JSON output of a voterInfoQuery to the Google Civic Information
+   *     API
+   * @return the new Election object
+   */
+  public static Election fromVoterInfoQuery(
```

Comment:
I think this method is probably better off being **not** static.  Then you don't need to pass in the Election object, and can instead call it ON the election object.  E.g.
Right now: `Election.fromVoterInfoQuery(election, datastore, voterInfoQueryData)`
Not static: `election.addVoterInfoQueryData(datastore, voterInfoQueryData)`

Reply:
+1